discovery-infra/tests/test_agent.py

```diff
@@ -0,0 +1,30 @@
+import logging
+import pytest
+import waiting
+from tests.base_test import BaseTest
+
+
+class TestAgent(BaseTest):
+
+    @pytest.mark.regression
+    def test_kill_agent(self, api_client, node_controller):
+        # start vms, kill agent, validate it was restarted and works
+
+        # Define new cluster
+        cluster_id = self.create_cluster(api_client=api_client).id
+        # Generate and download cluster ISO
+        self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
+        # Boot nodes into ISO
+        hosts = node_controller.start_all_nodes()
+        # kill agent
+        test_host = list(hosts.values())[0]
+        test_host.kill_service("agent")
+        # wait till agent is up
```

Comment:
I think you should wait till it is up before you kill it, otherwise you won't know you actually killed it.

Reply:
optional